File: DiKErnel/Test/DiKErnel.Integration.TestUtil/GrassRevetmentWaveRunupLocationDependentInputAssertHelper.cpp
// Copyright (C) Stichting Deltares 2022. All rights reserved.
//
// This file is part of DiKErnel.
//
// DiKErnel is free software: you can redistribute it and/or modify
// it under the terms of the GNU Lesser General Public License as published by
// the Free Software Foundation, version 3.
// 
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
// GNU Lesser General Public License for more details.
//
// You should have received a copy of the GNU Lesser General Public License
// along with this program. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.
//
// All names, logos, and references to "Deltares" are registered trademarks of
// Stichting Deltares and remain full property of Stichting Deltares at all times.
// All rights reserved.

#include "GrassRevetmentWaveRunupLocationDependentInputAssertHelper.h"

#include <gtest/gtest.h>

namespace DiKErnel::Integration::TestUtil
{
    using namespace std;

    void GrassRevetmentWaveRunupLocationDependentInputAssertHelper::AssertGeneralProperties(
        const double x,
        const double outerSlope,
        const GrassRevetmentWaveRunupLocationDependentInput& locationDependentInput)
    {
        ASSERT_DOUBLE_EQ(x, locationDependentInput.GetX());
        ASSERT_DOUBLE_EQ(outerSlope, locationDependentInput.GetOuterSlope());
    }

    void GrassRevetmentWaveRunupLocationDependentInputAssertHelper::AssertRepresentative2P(
        const double representative2PAru,
        const double representative2PBru,
        const double representative2PCru,
        const double gammab,
        const double gammaf,
        const GrassRevetmentWaveRunupRepresentative2P& representative2P)
    {
        ASSERT_DOUBLE_EQ(representative2PAru, representative2P.GetRepresentative2PAru());
        ASSERT_DOUBLE_EQ(representative2PBru, representative2P.GetRepresentative2PBru());
        ASSERT_DOUBLE_EQ(representative2PCru, representative2P.GetRepresentative2PCru());
        ASSERT_DOUBLE_EQ(gammab, representative2P.GetGammab());
        ASSERT_DOUBLE_EQ(gammaf, representative2P.GetGammaf());
    }

    void GrassRevetmentWaveRunupLocationDependentInputAssertHelper::AssertWaveAngleImpact(
        const double abeta,
        const double betamax,
        const GrassRevetmentWaveRunupWaveAngleImpact& waveAngleImpact)
    {
        ASSERT_DOUBLE_EQ(abeta, waveAngleImpact.GetAbeta());
        ASSERT_DOUBLE_EQ(betamax, waveAngleImpact.GetBetamax());
    }

    void GrassRevetmentWaveRunupLocationDependentInputAssertHelper::AssertCumulativeOverload(
        const double criticalCumulativeOverload,
        const GrassRevetmentWaveRunupLocationDependentInput& locationDependentInput)
    {
        ASSERT_DOUBLE_EQ(criticalCumulativeOverload, locationDependentInput.GetCriticalCumulativeOverload());
    }

    void GrassRevetmentWaveRunupLocationDependentInputAssertHelper::AssertFrontVelocity(
        const double criticalFrontVelocity,
        const GrassRevetmentWaveRunupLocationDependentInput& locationDependentInput)
    {
        ASSERT_DOUBLE_EQ(criticalFrontVelocity, locationDependentInput.GetCriticalFrontVelocity());
    }

    void GrassRevetmentWaveRunupLocationDependentInputAssertHelper::AssertTransitionAlpha(
        const double increasedLoadTransitionAlphaM,
        const double reducedStrengthTransitionAlphaS,
        const GrassRevetmentWaveRunupLocationDependentInput& locationDependentInput)
    {
        ASSERT_DOUBLE_EQ(increasedLoadTransitionAlphaM, locationDependentInput.GetIncreasedLoadTransitionAlphaM());
        ASSERT_DOUBLE_EQ(reducedStrengthTransitionAlphaS, locationDependentInput.GetReducedStrengthTransitionAlphaS());
    }

    void GrassRevetmentWaveRunupLocationDependentInputAssertHelper::AssertAverageNumberOfWaves(
        const double averageNumberOfWavesCtm,
        const GrassRevetmentWaveRunupLocationDependentInput& locationDependentInput)
    {
        ASSERT_DOUBLE_EQ(averageNumberOfWavesCtm, locationDependentInput.GetAverageNumberOfWavesCtm());
    }
}

File: Test/DiKErnel.Integration.TestUtil/GrassRevetmentWaveRunupLocationDependentInputAssertHelper.cs
// Copyright (C) Stichting Deltares and State of the Netherlands 2023. All rights reserved.
//
// This file is part of DiKErnel.
//
// This program is free software; you can redistribute it and/or modify it under the terms of
// the GNU Lesser General Public License as published by the Free Software Foundation; either
// version 3 of the License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
// without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
// See the GNU Lesser General Public License for more details.
//
// You should have received a copy of the GNU Lesser General Public License along with this
// program; if not, see <https://www.gnu.org/licenses/>.
//
// All names, logos, and references to "Deltares" are registered trademarks of Stichting
// Deltares and remain full property of Stichting Deltares at all times. All rights reserved.

using DiKErnel.Integration.Data.GrassRevetmentWaveRunup;
using NUnit.Framework;

namespace DiKErnel.Integration.TestUtil
{
    internal static class GrassRevetmentWaveRunupLocationDependentInputAssertHelper
    {
        public static void AssertGeneralProperties(double outerSlope,
                                                   GrassRevetmentWaveRunupLocationDependentInput locationDependentInput)
        {
            Assert.That(locationDependentInput.OuterSlope, Is.EqualTo(outerSlope));
        }

        public static void AssertRepresentative2P(double representative2PAru, double representative2PBru,
                                                  double representative2PCru, double gammab, double gammaf,
                                                  GrassRevetmentWaveRunupRepresentative2P representative2P)
        {
            Assert.That(representative2P.Representative2PAru, Is.EqualTo(representative2PAru));
            Assert.That(representative2P.Representative2PBru, Is.EqualTo(representative2PBru));
            Assert.That(representative2P.Representative2PCru, Is.EqualTo(representative2PCru));
            Assert.That(representative2P.Gammab, Is.EqualTo(gammab));
            Assert.That(representative2P.Gammaf, Is.EqualTo(gammaf));
        }

        public static void AssertWaveAngleImpact(double abeta, double betamax,
                                                 GrassRevetmentWaveRunupWaveAngleImpact waveAngleImpact)
        {
            Assert.That(waveAngleImpact.Abeta, Is.EqualTo(abeta));
            Assert.That(waveAngleImpact.Betamax, Is.EqualTo(betamax));
        }

        public static void AssertCumulativeOverload(double criticalCumulativeOverload,
                                                    GrassRevetmentWaveRunupLocationDependentInput locationDependentInput)
        {
            Assert.That(locationDependentInput.CriticalCumulativeOverload, Is.EqualTo(criticalCumulativeOverload));
        }

        public static void AssertFrontVelocity(double criticalFrontVelocity,
                                               GrassRevetmentWaveRunupLocationDependentInput locationDependentInput)
        {
            Assert.That(locationDependentInput.CriticalFrontVelocity, Is.EqualTo(criticalFrontVelocity));
        }

        public static void AssertTransitionAlpha(double increasedLoadTransitionAlphaM,
                                                 double reducedStrengthTransitionAlphaS,
                                                 GrassRevetmentWaveRunupLocationDependentInput locationDependentInput)
        {
            Assert.That(locationDependentInput.IncreasedLoadTransitionAlphaM, Is.EqualTo(increasedLoadTransitionAlphaM));
            Assert.That(locationDependentInput.ReducedStrengthTransitionAlphaS, Is.EqualTo(reducedStrengthTransitionAlphaS));
        }

        public static void AssertAverageNumberOfWaves(double averageNumberOfWavesCtm,
                                                      GrassRevetmentWaveRunupLocationDependentInput locationDependentInput)
        {
            Assert.That(locationDependentInput.AverageNumberOfWavesCtm, Is.EqualTo(averageNumberOfWavesCtm));
        }
    }
}